elastic/elastic-agent-shipper
# Kafka integration test

## What does this PR do?

This PR adds integration tests for the Kafka output, with the use cases, and some of the mechanics around reading events from a kafka topic ported from the beats output. 

This PR also includes infrastructure to create a test docker image with Kafka, exposing ports to allow testing over plaintext and SSL.


## Checklist


~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added an entry in `CHANGELOG.md` or `CHANGELOG-developer.md`.~


## How to test this PR locally

- Run `docker-compose up` in the `testing/environment/docker` folder
- Run `mage integrationTest`

Note that I needed to delete the `docker_kafka` container after each test run to allow the tests to continue to run.


## Related issues

Relates #166, #176
